ledger/src/shred.rs

```diff
@@ -467,8 +473,36 @@ impl Shred {
         self.common_header.signature
     }
 
-    pub fn seed(&self) -> [u8; 32] {
+    pub fn seed(
+        &self,
+        leader_schedule_cache: &Arc<LeaderScheduleCache>,
+        bank_forks: &Arc<RwLock<BankForks>>,
+    ) -> [u8; 32] {
         let mut seed = [0; 32];
+
+        let bf = bank_forks.read().unwrap();
+        let maybe_bank = bf.get(self.slot());
```

Comment:
I think similar to the previous feature-gate that we did, checking for feature activation should use the root-bank:
https://github.com/solana-labs/solana/blob/f2ab0384e/core/src/retransmit_stage.rs#L276-L290
So instead: 
```
let root_bank = bank_forks.read().unwrap().root_bank();
```

Reply:
changed